mithril-aggregator/src/http_server/routes/aggregator_status.rs

```diff
@@ -0,0 +1,284 @@
+use warp::Filter;
+
+use mithril_common::{messages::AggregatorStatusMessage, StdResult};
+
+use crate::{
+    dependency_injection::EpochServiceWrapper,
+    http_server::routes::{middlewares, router::RouterState},
+};
+
+pub fn routes(
+    router_state: &RouterState,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = warp::Rejection> + Clone {
+    status(router_state)
+}
+
+/// GET /status
+fn status(
+    router_state: &RouterState,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = warp::Rejection> + Clone {
+    warp::path!("status")
+        .and(warp::get())
+        .and(middlewares::with_logger(router_state))
+        .and(middlewares::with_epoch_service(router_state))
+        .and(middlewares::extract_config(router_state, |config| {
+            config.cardano_node_version.clone()
+        }))
+        .and_then(handlers::status)
+}
+
+async fn get_aggregator_status_message(
+    epoch_service: EpochServiceWrapper,
+    cardano_node_version: String,
+) -> StdResult<AggregatorStatusMessage> {
+    let epoch_service = epoch_service.read().await;
+
+    let epoch = epoch_service.epoch_of_current_data()?;
+    let cardano_era = epoch_service.cardano_era()?;
+    let mithril_era = epoch_service.mithril_era()?;
+    let aggregator_node_version = env!("CARGO_PKG_VERSION").to_string();
+    let protocol_parameters = epoch_service.current_protocol_parameters()?.clone();
+    let next_protocol_parameters = epoch_service.next_protocol_parameters()?.clone();
+    let total_signers = epoch_service.current_signers()?.len();
+    let total_next_signers = epoch_service.next_signers()?.len();
+    let total_stakes_signers = epoch_service
+        .current_signers_with_stake()?
+        .iter()
+        .map(|s| s.stake)
+        .sum();
+    let total_next_stakes_signers = epoch_service
+        .next_signers_with_stake()?
+        .iter()
+        .map(|s| s.stake)
+        .sum();
```

Comment:
We don't want to have these computations done at every call of the route, they should be done once per epoch.

Reply:
This is now computed in [`inform_epoch`](https://github.com/input-output-hk/mithril/blob/30c8cc825c6f3b2f0f3e6bfa6049c97a14148063/mithril-aggregator/src/services/epoch_service.rs#L344-L345) once per epoch.